instrumentation/trilogy/opentelemetry-instrumentation-trilogy.gemspec

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+lib = File.expand_path('lib', __dir__)
+$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
+require 'opentelemetry/instrumentation/trilogy/version'
+
+Gem::Specification.new do |spec|
+  spec.name        = 'opentelemetry-instrumentation-trilogy'
+  spec.version     = OpenTelemetry::Instrumentation::Trilogy::VERSION
+  spec.authors     = ['OpenTelemetry Authors']
+  spec.email       = ['cncf-opentelemetry-contributors@lists.cncf.io']
+
+  spec.summary     = 'Trilogy instrumentation for the OpenTelemetry framework'
+  spec.description = 'Adds auto instrumentation that includes SQL metadata'
+  spec.homepage    = 'https://github.com/open-telemetry/opentelemetry-ruby'
+  spec.license     = 'Apache-2.0'
+
+  spec.files = ::Dir.glob('lib/**/*.rb') +
+               ::Dir.glob('*.md') +
+               ['LICENSE', '.yardopts']
+  spec.require_paths = ['lib']
+  spec.required_ruby_version = Gem::Requirement.new('>= 2.5.0')
+
+  spec.add_dependency 'opentelemetry-api', '~> 1.0'
+  spec.add_dependency 'opentelemetry-instrumentation-base', '~> 0.19.0'
```

Comment:
I believe we need to require the semantic conventions gem as a dependency here actually. It's required by the `-sdk` gem but not by the `-api` or `-base`. I think the appropriate way to version it is loosely, ala what the the `-sdk` has done https://github.com/open-telemetry/opentelemetry-ruby/blob/e639d20fbcc355f1df481b9867b922a83c481050/sdk/opentelemetry-sdk.gemspec#L30

Upon inspection of other instrumentations, I think we also need to fix the recent Aws-Sdk instrumentation gem, but I can do that in a seperate PR 😬 😅

Reply:
Makes sense. I will fix this. 

Dang, I wish there was a way I could have caught that earlier!!!

Reply:
hmmm... looks to me like _none_ of the instrumentations explicitly add the conventions as a dependency and rely on the local path from the Gemfile. 🤔 

I guess we do not have many uses cases were folks are using the instrumentations _without_ the SDK. 

```console
omoplata ~/github/opentelemetry-ruby(arielvalentin/trilogy-instrumentation|✔) %
🤘 rg "add_dependency.*opentelemetry-semantic_conventions" -g '*.gemspec'
sdk/opentelemetry-sdk.gemspec
36:  spec.add_dependency 'opentelemetry-semantic_conventions'
```

```console
omoplata ~/github/opentelemetry-ruby(arielvalentin/trilogy-instrumentation|✔) %
🤘 rg "opentelemetry-semantic_conventions" -g 'Gemfile'

resource_detectors/Gemfile:gem 'opentelemetry-semantic_conventions', path: '../semantic_conventions'
exporter/zipkin/Gemfile:gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
exporter/jaeger/Gemfile:gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
exporter/otlp/Gemfile:gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
sdk/Gemfile:gem 'opentelemetry-semantic_conventions', path: '../semantic_conventions'
instrumentation/concurrent_ruby/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/resque/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/ethon/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/pg/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/delayed_job/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/base/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/excon/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/rails/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/faraday/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/rspec/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/sinatra/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/redis/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/active_job/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/koala/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/restclient/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/mongo/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/action_pack/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/rack/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/net_http/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/trilogy/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/graphql/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/http/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/active_support/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/lmdb/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/all/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/active_record/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/aws_sdk/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/bunny/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/http_client/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/active_model_serializers/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/que/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/ruby_kafka/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/mysql2/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/action_view/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/sidekiq/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
instrumentation/dalli/Gemfile:  gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
propagator/ottrace/Gemfile:gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
propagator/jaeger/Gemfile:gem 'opentelemetry-semantic_conventions', path: '../../semantic_conventions'
```

Reply:
I think basically The resource detector and aws-sdk instrumentation, afaik, are the only gems that directly reference the SemanticConventions module . The resource detector depends on the sdk in the gemspec tho, which in turn depends on the semconv get, while instrumentations (or -base) do not. I think the search above is only pulling in the :test dependencies from the Gemfile.

Any way tldr looks good to me now